File: server/src/main/java/com/ecommerce/ecommercerestapi/Entity/UserPayment.java
package com.ecommerce.ecommercerestapi.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "user_payment")
public class UserPayment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @ManyToOne(cascade = { CascadeType.PERSIST, CascadeType.MERGE })
    private User user;

    @Column(name = "payment_type")
    private String paymentType;

    @Column(name = "account_number")
    private String accountNumber;

    private String valid;

    private String since;

    private String cvv;

    private Boolean active;
}

File: client/src/interfaces/UserPayment.ts
import { User } from "./User";

export interface UserPaymentReq {
  id?: number;
  user: {
    id: number;
  };
  accountNumber: string;
  active?: boolean;
  paymentType: string;
  cvv: string;
  since: string;
  valid: string;
}

export interface UserPaymentRes {
  id?: number;
  user: User;
  accountNumber: string;
  active: boolean;
  paymentType: string;
  cvv: string;
  since: string;
  valid: string;
}

export const INITIAL_USER_PAYMENT_REQ = {
  user: {
    id: 0,
  },
  accountNumber: "",
  active: true,
  paymentType: "",
  cvv: "",
  since: "",
  valid: "",
  id: 0,
};
